pkg/utils/kubernetes/reduce/filters.go

```diff
@@ -94,24 +94,56 @@ func filterClusterRoles(clusterRoles []rbacv1.ClusterRole) []rbacv1.ClusterRole
 	return append(clusterRoles[:best], clusterRoles[best+1:]...)
 }
 
+// -----------------------------------------------------------------------------
+// Filter functions - Roles
+// -----------------------------------------------------------------------------
+
+// filterRoles filters out the Role to be kept and returns
+// all the Roles to be deleted.
+// The filtered-out Role is decided as follows:
+//  1. creationTimestamp (newer is better, because newer Roles can contain new policy rules)
+func filterRoles(roles []rbacv1.Role) []rbacv1.Role {
+	if len(roles) == 1 {
+		return []rbacv1.Role{}
+	}
+
+	best := 0
+	for i, cr := range roles {
+		if cr.CreationTimestamp.After(roles[best].CreationTimestamp.Time) {
+			best = i
+		}
+	}
+
+	return append(roles[:best], roles[best+1:]...)
+}
+
 // -----------------------------------------------------------------------------
 // Filter functions - ClusterRoleBindings
 // -----------------------------------------------------------------------------
 
-// filterClusterRoleBindings filters out the ClusterRoleBinding to be kept and returns
-// all the ClusterRoleBindings to be deleted.
-// The filtered-out ClusterRoleBinding is decided as follows:
-// 1. creationTimestamp (older is better)
-func filterClusterRoleBindings(clusterRoleBindings []rbacv1.ClusterRoleBinding) []rbacv1.ClusterRoleBinding {
+// filterBindings filters out the ClusterRoleBinding or RoleBindings to be kept and returns
+// all the objects to be deleted.
+func filterBindings[
```

Comment:
```suggestion
func filterRoleBindings[
```

I think that explicitly stating we are talking about roles can help to quickly understand what this function is about.

Reply:
95c8d45a6d6fecad44a59ffea5990f475b4c38f6